### PR TITLE
Custom contracts job

### PR DESCRIPTION
**Pull request by nguyer**
Adds a test job for custom contracts

Depends on https://github.com/hyperledger/firefly-cli/issues/151 and https://github.com/hyperledger/firefly/pull/589

### CODE REVIEW

File: internal/perf/custom_contract.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	req := pr.client.R().
+		SetHeaders(map[string]string{
+			"Accept":       "application/json",
+			"Content-Type": "application/json",
+		}).
+		SetBody([]byte(payload))
+	pr.sendAndWait(req, "contracts/invoke", id, conf.PerfCmdCustomContract.String())
+}
+
+func (pr *perfRunner) GetBlockchainEventValue(ref string) (int, error) {
```
[the remark this answers is not in the document]
Comment (nguyer) [reply internal/perf/custom_contract.go:44]: Not currently. In an upcoming PR there will be, and I can update this after that.

File: internal/perf/perf.go [hunk 2]
```diff
@@ -188,6 +224,9 @@ func (pr *perfRunner) sendAndWait(req *resty.Request, ep string, id int, action
 				json.Unmarshal(res.Body(), &tokenRes)
 				pr.updateMsgTime(tokenRes.LocalID.String())
 				log.Infof("%d --> %s Sent with Token ID: %s", id, action, tokenRes.LocalID)
+			case conf.PerfCmdCustomContract.String():
+				json.Unmarshal(res.Body(), &contractRes)
+				log.Infof("%d --> Invoked contract", id)
```
[the remark this answers is not in the document]
Comment (nguyer) [reply internal/perf/perf.go:229]: There's an operation ID but there's nothing to correlate that to the expected event that will be emitted. We don't do anything with the operation ID at this point.

File: internal/perf/perf.go [hunk 3]
```diff
@@ -147,18 +158,42 @@ func (pr *perfRunner) eventLoop() (err error) {
...
+					log.Infof("\n\t%d - Received \n\t%d --- Worker ID: %s\n\t%d --- Ref: %s", workerId, workerId, event.ID.String(), workerId, event.Reference)
+				}
+			default:
+				workerId, err = strconv.Atoi(strings.ReplaceAll(event.Message.Header.Tag, pr.tagPrefix+"_", ""))
+				if err != nil {
+					log.Errorf("Could not parse message tag: %s", err)
+					continue
+				}
+				pr.deleteMsgTime(event.Message.Header.ID.String())
+				log.Infof("\n\t%d - Received \n\t%d --- Event ID: %s\n\t%d --- Message ID: %s", workerId, workerId, event.ID.String(), workerId, event.Message.Header.ID.String())
```
[the remark this answers is not in the document]
Comment (nguyer) [reply internal/perf/perf.go:180]: For the default case, what is the difference? The existing code logged "Messages ID" as the worker ID. It feels like some of these things are redundant.

[the remark this answers is not in the document]
Comment (nguyer) [reply internal/perf/perf.go:180]: Ah, the worker ID is the goroutine index. The confusing part is the custom contract log labels what is really the event ID as a worker ID. I'll fix that.

File: internal/perf/custom_contract.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (eberger727) [reply internal/perf/custom_contract.go:44]: https://github.com/hyperledger/firefly-perf-cli/issues/14

**Pull request merged**